File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/KActionType.ts

class KActionType {

  public static SET: KActionType = new KActionType("SET");
  public static ADD: KActionType = new KActionType("ADD");
  public static REMOVE: KActionType = new KActionType("DEL");
  public static NEW: KActionType = new KActionType("NEW");
  private code: string = "";

  constructor(code: string) {
    this.code = code;
  }

  public toString(): string {
    return this.code;
  }

  public equals(other: AccessMode): boolean {
        return this == other;
    }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/KActionType.java
package org.kevoree.modeling;

public enum KActionType {

    CALL("CALL"),
    CALL_RESPONSE("CALL_RESPONSE"),
    SET("SET"),
    ADD("ADD"),
    REMOVE("DEL"),
    NEW("NEW");

    private String _code = "";

    KActionType(String code) {
        this._code = code;
    }

    public String toString() {
        return _code;
    }

    public String code(){return _code;}

    public static KActionType parse(String s) {
        for(int i = 0; i < values().length; i++) {
            KActionType current = values()[i];
            if(current.code().equals(s)) {
                return current;
            }
        }
        return null;
    }

}